shopify_function_macro/src/lib.rs

```diff
@@ -127,14 +128,73 @@ pub fn generate_types(attr: proc_macro::TokenStream) -> proc_macro::TokenStream
         #[graphql(
             query_path = #OUTPUT_QUERY_FILE_NAME,
             schema_path = #schema_path,
-            response_derives = "Clone,Debug,PartialEq,Eq,Deserialize",
-            variables_derives = "Clone,Debug,PartialEq,Eq,Deserialize",
+            response_derives = "Clone,Debug,PartialEq,Eq,Deserialize,OneOfDefault",
+            variables_derives = "Clone,Debug,PartialEq,Eq,Deserialize,OneOfDefault",
             skip_serializing_none
         )]
         struct Output;
     }
     .into()
 }
 
+fn is_type(ty: &syn::Type, name: impl AsRef<str>) -> bool {
+    if let syn::Type::Path(path) = ty {
+        path.path.segments.len() == 1
+            && path
+                .path
+                .segments
+                .first()
+                .unwrap()
+                .ident
+                .to_string()
+                .as_str()
```

Comment:
I think we can make use of [this `PartialEq` impl](https://docs.rs/syn/latest/syn/struct.Ident.html#impl-PartialEq%3CT%3E-for-Ident):
```suggestion
```